src/puppetlabs/puppetdb/query_eng/engine.clj

```diff
@@ -1789,7 +1810,8 @@
               (map->NullExpression {:column cinfo :null? value}))
 
             [["~" column-name value]]
-            (let [cinfo (get-in query-rec [:projections column-name])]
+            (let [colname (first (str/split column-name #"\."))
```

Comment:
What will we do for parameters.foo.bar ~ "baz"? Does it look for a map as foo's value and the bar key inside there?